app/views/projects/blob/_editor.html.haml

```diff
@@ -17,7 +17,10 @@
 
   .file-content.code
     %pre.js-edit-mode-pane#editor
-      = params[:content] || local_assigns[:blob_data]
+      - content = params[:content] || local_assigns[:blob_data]
+      = content
+      - if content.length && 0 === content[content.length-1].index(/[\r\n]/)
```

Comment:
It do what this issues are about: preserve the trailing end of line (avoid the auto-trim in previous version and now avoid a line to be added automatically). In short word, you will not have the end of the file modified if you haven't added manually: if the last char is a new line, re-add it: https://github.com/gitlabhq/gitlabhq/pull/9422